core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java

```diff
@@ -63,9 +64,16 @@ public String getName() {
 
         @Override
         public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient client) throws IOException {
+            validateOpType(request.params().get("op_type"));
             request.params().put("op_type", "create");
             return RestIndexAction.this.prepareRequest(request, client);
         }
+
+        public void validateOpType(String opType) {
+            if (null != opType && !"create".equals(opType.toLowerCase(Locale.US))) {
```

Comment:
please use the root locale

Reply:
also please use `== false` for comparison

Reply:
@s1monw I just made the requested changes.